src/main/java/bio/terra/service/dataset/StorageResourceDao.java

```diff
@@ -73,11 +75,18 @@ public StorageResourceDao(DataRepoJdbcConfiguration jdbcConfiguration) {
 
     private static class StorageResourceMapper implements RowMapper<StorageResource> {
         public StorageResource mapRow(ResultSet rs, int rowNum) throws SQLException {
-            return new StorageResource()
-                .datasetId(UUID.fromString(rs.getString("dataset_id")))
-                .cloudPlatform(CloudPlatform.valueOf(rs.getString("cloud_platform")))
-                .cloudResource(GoogleCloudResource.valueOf(rs.getString("cloud_resource")))
-                .region(GoogleRegion.valueOf(rs.getString("region")));
+            final CloudPlatform cloudPlatform = CloudPlatform.valueOf(rs.getString("cloud_platform"));
+            switch (cloudPlatform) {
```

Comment:
It's a shame that `CloudPlatform` is auto generated, this would be a great case for a enum method.